pkg/api/types.go

```diff
@@ -705,6 +713,8 @@ type ServiceSpec struct {
 	// not be changed by updates.
 	PortalIP string `json:"portalIP,omitempty"`
 
+	Rewrite bool `json:"rewrite,omitempty" description:"Does the load balancer rewrite packets or pass them through"`
```

Comment:
Yes, that's exactly what I was thinking.

The main hassle, then, is configuring the kube-proxy to know what the cloudprovider is, but I think it's a better approach.